File: atcoder/d154.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

double sum(ll n){
    return (n+1)/(double) 2;
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    ll n, k;
    cin >> n >> k;
    vector <ll> arr(n);
    for(ll i = 0; i < n; i++){
        cin >> arr[i];
    }
    ll start = 0, end = start+k-1;
    ll maxsum = INT_MIN, curr_sum = 0, maxstart = 0;
    for(ll i = 0; i < k; i++){
        curr_sum += arr[i];
    }
    while(true){
        if(curr_sum > maxsum){
            maxsum = curr_sum;
            maxstart = start;
        }
        if(end == n-1) break;
        curr_sum -= arr[start];
        curr_sum += arr[end+1];
        end++;
        start++;
    }
    double ret = 0;
    for(ll i = maxstart; i < maxstart+k; i++){
        ret += sum(arr[i]);
    }
    cout << fixed << setprecision(6) << ret << endl;

}

File: atcoder/d154.py
def sum_n(n):
    return (n*(n+1))//2

nk = [int(s) for s in input().split()]
n, k = nk[0], nk[1]
arr = [int(s) for s in input().split()]
if k == 0:
    print(0)
else:
    maxsum = -100000
    currsum = 0
    maxstart = 0
    start, end = 0, k-1
    for i in range(k):
        currsum += arr[k]
    while True:
        if currsum > maxsum:
            maxsum = currsum
            maxstart = start
        if end == n-1:
            break
        currsum += arr[end+1]
        currsum -= arr[start]
        start += 1
        end += 1
    ret = 0
    for i in range(maxstart, maxstart+k):
        if arr[i] == 0:
            ret += 0
        else : ret += sum_n(arr[i])/arr[i]
    print(ret)
